packages/rest/src/rest.server.ts

```diff
@@ -281,7 +282,13 @@ export class RestServer extends Context implements Server, HttpServerLike {
     // See https://github.com/strongloop/loopback-next/issues/433
     if (this._httpHandler) return;
 
-    this._httpHandler = new HttpHandler(this);
+    /**
+     * Check if there is custom router in the context
+     */
+    const router = this.getSync(RestBindings.ROUTER, {optional: true});
+    const routingTable = new RoutingTable(router);
+
+    this._httpHandler = new HttpHandler(this, routingTable);
```

Comment:
I am wondering if it's a good idea for HttpHandler to accept RoutingTable instance. Have you considered an alternative design where HttpHandler accepts a router instance only?

```ts
const router = this.getSync(RestBindings.ROUTER, {optional: true});
this._httpHandler = new HttpHandler(this, router);
```

IMO, the RoutingTable class is an implementation detail of HttpHandler that's not open for extensibility to consumers.

Reply:
Please note that using the constructor the pass down a version of RestRouter is still internal implementation details, which we can change later. The public contract is to bind a custom RestRouter to the given key.